src/core/column/cumminmax.h

```diff
@@ -57,24 +57,42 @@ class CumMinMax_ColumnImpl : public Virtual_ColumnImpl {
         [&](size_t gi) {
           size_t i1 = size_t(offsets[gi]);
           size_t i2 = size_t(offsets[gi + 1]);
-
+        if (REVERSE) {
           T val;
-          bool res_valid = col_.get_element(i1, &val);
-          data[i1] = res_valid? val : GETNA<T>();
-
-          for (size_t i = i1 + 1; i < i2; ++i) {
+          bool res_valid = col_.get_element(i2-1, &val);
+          data[i2-1] = res_valid? val : GETNA<T>();
+          for (size_t i = i2-1; i-- > i1;) {
             bool is_valid = col_.get_element(i, &val);
             if (is_valid) {
               if (MIN) {
-                data[i] = (res_valid && data[i - 1] < val)? data[i - 1] : val;
+                data[i] = (res_valid && data[i + 1] < val)? data[i + 1] : val;
               } else {
-                data[i] = (res_valid && data[i - 1] > val)? data[i - 1] : val;
+                data[i] = (res_valid && data[i + 1] > val)? data[i + 1] : val;
               }
               res_valid = true;
             } else {
-              data[i] = data[i - 1];
+              data[i] = data[i + 1];
+            }
             }
-          }
+        } else {
+            T val;
+            bool res_valid = col_.get_element(i1, &val);
+            data[i1] = res_valid? val : GETNA<T>();
+  
+            for (size_t i = i1 + 1; i < i2; ++i) {
+              bool is_valid = col_.get_element(i, &val);
+              if (is_valid) {
+                if (MIN) {
+                  data[i] = (res_valid && data[i - 1] < val)? data[i - 1] : val;
+                } else {
+                  data[i] = (res_valid && data[i - 1] > val)? data[i - 1] : val;
+                }
+                res_valid = true;
+              } else {
+                data[i] = data[i - 1];
+              }
+            }}
```

Comment:
Same here. We normally align the opening and closing brackets.

Reply:
I'm using Sublime and 2-space indentation for C++ / 4-space indentation for Python. Then, once I press the tab button I get the required indentation depending on the language. As for the brackets, Sublime automatically indents the closing bracket (using the appropriate number of spaces) once you type in the opening one.